xwayland/xwm.c

```diff
@@ -1215,6 +1233,19 @@ static void xwm_handle_client_message(struct wlr_xwm *xwm,
 		xwm_handle_wm_protocols_message(xwm, ev);
 	} else if (ev->type == xwm->atoms[NET_ACTIVE_WINDOW]) {
 		xwm_handle_net_active_window_message(xwm, ev);
+	} else if (ev->type == xwm->atoms[WM_CHANGE_STATE]) {
+		struct wlr_xwayland_surface *xsurface = lookup_surface(xwm, ev->window);
```

Comment:
Can we extract all of this into a `xwm_handle_wm_change_state_message` function?

Reply:
Can you look into this?